src/main/java/main/Game.java

```diff
@@ -139,16 +139,16 @@ private void init() throws Exception {
 	}
 
 	private void setIcon() {
+		TextureInfo textureInfo = OpenGLUtils.loadTexture("/icons/soldier.png");
+		ByteBuffer buffer = textureInfo.buffer;
 		GLFWImage image = GLFWImage.malloc();
-		TextureInfo texture = OpenGLUtils.loadTexture("/icons/soldier.png");
-		ByteBuffer temp = texture.buffer;
-		image.set(texture.width, texture.height, temp);
+		image.set(textureInfo.width, textureInfo.height, buffer);
 		GLFWImage.Buffer images = GLFWImage.malloc(1);
 		images.put(0, image);
 		glfwSetWindowIcon(window, images);
+
 		images.free();
 		image.free();
-
```

Comment:
This changes nothing?

Reply:
the buffer was called temp but it wasn't temp

Reply:
Fair enough but you change way more than that